src/sprightly.ts

```diff
@@ -1,60 +1,85 @@
-import { readFile } from "fs/promises"
+import { readFileSync } from "fs"
+import path from "path"
+import KeyNotFoundError from "./errors/KeyNotFoundError"
+import PathNotFoundError from "./errors/PathNotFoundError"
 
 interface Data {
   [key: string]: string | number | Array<string | number | Data> | Data
 }
 
-type Options = {
-  cache?: boolean
+interface Options {
   keyFallback?: string
   throwOnKeyNotfound?: boolean
+  cache?: boolean
 }
 
-const defaultOptions: Options = {
+const cache = new Map<string, string>()
+
+const defaultOptions = {
   keyFallback: "",
   throwOnKeyNotfound: false,
   cache: false,
 }
 
-export = async function sprightly(
+function parse(file: string, data: Data, options: Options) {
+  const parsedFile = file.replaceAll(
+    /\{\{(>?)(.*?)\}\}/g,
+    (_, isComponent: string, reference: string) => {
+      if (isComponent) {
+        try {
+          return sprightly(reference, data, options)
+        } catch (e) {
+          throw new PathNotFoundError(reference.trim())
+        }
+      }
+      return reference
+        .split(/((?:\w+))(?:(?:\[)(\d+)(?:\]))?(?:\.?)/gm)
+        .filter((v) => v && Boolean(v.trim()))
+        .reduce((obj, key) => {
+          if (
+            typeof obj === "object" &&
```

Comment:
Why do you need to check if this is an object?

Reply:
If it is a primitive property, and I use `in` against it, it will throw an error